simplesat/pool.py

```diff
@@ -47,7 +49,10 @@ def add_repository(self, repository):
             self._id += 1
             self._id_to_package_[current_id] = package
             self._package_to_id_[package] = current_id
-            self._packages_by_name_[package.name].append(package)
+            for constraints in package.provides:
+                req = Requirement.from_constraints(constraints)
+                assert not req.has_any_version_constraint
```

Comment:
Ah yes, I forgot to write about it. Really we need to move the entire package metadata spec into the docs here.

The long and short of it is that right now we don't have a good way to handle provides with version numbers. We might someday, but there aren't any good use-cases that I know of. This assertion is there to make sure package metadata authors haven't done that. It should probably be `ValueError` or `InvalidConstraint` or something instead. What do you think?